aws/internal/service/apigatewayv2/waiter/waiter.go

```diff
@@ -0,0 +1,32 @@
+package waiter
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/apigatewayv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+const (
+	// Maximum amount of time to wait for a Deployment to return Deployed
+	DeploymentDeployedTimeout = 5 * time.Minute
+)
+
+// DeploymentDeployed waits for a Deployment to return Deployed
+func DeploymentDeployed(conn *apigatewayv2.ApiGatewayV2, apiId, deploymentId string) (*apigatewayv2.GetDeploymentOutput, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{apigatewayv2.DeploymentStatusPending},
+		Target:  []string{apigatewayv2.DeploymentStatusDeployed},
+		Refresh: DeploymentStatus(conn, apiId, deploymentId),
+		Timeout: DeploymentDeployedTimeout,
+	}
+
+	outputRaw, err := stateConf.WaitForState()
+
+	switch output := outputRaw.(type) {
```

Comment:
Since there are only two possibilities, would `if v, ok := outputRaw.(*apigatewayv2.GetDeploymentOutput); ok { ... }` work better here?

Reply:
I based the code on the original PR - https://github.com/terraform-providers/terraform-provider-aws/pull/12765/files#diff-51328917bb577dc7dd98f1ece91ed566R26-R31 but have no strong opinion.

Reply:
On implementation, your suggestions seems more maintainable.
I made the change.

Reply:
Brian has updated his code in that PR to use the `if` version